site/oauth2-examples-proxy.md

```diff
@@ -0,0 +1,75 @@
+# Use OAuth2 Proxy and Keycloak as OAuth 2.0 server
+
+Let's test the following flow:
+
+* Access management ui via a browser thru OAuth2-Proxy
+
+<pre class="lang-plain">
+                    [ Keycloak ] 3. authenticate
+                      /|\  |
+                       |   | 4. token
+        2.redirect     |  \|/                                        [ RabbitMQ ]
+                [ Oauth2-Proxy ]       ----5. forward with token-->  [  http    ]
+                      /|\
+                       |
+            1. rabbit_admin from a browser
+</pre>
+
+## Prerequisites to follow this guide
+
+- Docker
+- make
+
+## Deploy Key Cloak
+
+First, deploy **Key Cloak**. It comes preconfigured with all the required scopes, users and clients.
+<pre class="bash">
+make start-keycloak
+</pre>
+
+**Key Cloak** comes configured with its own signing key. And [rabbitmq.conf](https://github.com/rabbitmq/rabbitmq-oauth2-tutorial/tree/main/conf/oauth2-proxy/rabbitmq.conf)
+is also configured with the same signing key.
+
+To access KeyCloak management interface go to http://0.0.0.0:8080/ and enter `admin` as username and password.
+
+There is a dedicated **KeyCloak realm** called `Test` configured as follows:
+
+* [rsa](http://0.0.0.0:8080/admin/master/console/#/realms/test/keys) signing key
+* [rsa provider](http://0.0.0.0:8080/admin/master/console/#/realms/test/keys/providers)
+* `rabbitmq-proxy-client` client
+
+## Start RabbitMQ
+
+To start RabbitMQ run the following two commands. The first one tells RabbitMQ to pick up the
+rabbit.conf found under [conf/oauth2-proxy/rabbitmq.conf](https://github.com/rabbitmq/rabbitmq-oauth2-tutorial/tree/main/conf/oauth2-proxy/rabbitmq.conf)
+
+<pre class="lang-plain">
+export MODE=oauth2-proxy
+make start-rabbitmq
+</pre>
+
+**NOTE**: Oauth2-proxy requires that the `aud` claim matches the client's id. However, RabbitMQ requires the
+`aud` field to match `rabbitmq` which is the designated `resource_server_id`. Given that it has been
+impossible to configure keycloak with both values, [rabbitmq.conf](https://github.com/rabbitmq/rabbitmq-oauth2-tutorial/tree/main/conf/oauth2-proxy/rabbitmq.conf) has
+the setting below which disables validation of the audience claim.
+
+<pre class="lang-ini">
+auth_oauth2.verify_aud = false
+</pre>
+
+
+## Start OAuth2 Proxy
+
+To start Oauth2-proxy we run the following command:
+
+<pre class="lang-plain">
+make start-oauth2-proxy
+</pre>
+
+Oauth2-proxy is configured using [Alpha configuration](https://github.com/rabbitmq/rabbitmq-oauth2-tutorial/tree/main/conf/oauth2-proxy/alpha-config.yaml). This type of configuration permits injecting the access token into the HTTP **Authorization** header.
```

Comment:
Again do they need to know the above? If so change the word "injecting" to "inserting" (if it makes sense)